scripts/ec2/schedule.coffee

```diff
@@ -0,0 +1,73 @@
+# Description:
+#   Provides conveniences around certain tags we want to manage
+#
+# Commands:
+#   hubot ec2 schedule <instance_id> [<instance_id> ...] --start=start_hour --stop=stop_hour Adds an appropriate Schedule tag to the instance(s). Start and stop default to 8 AM to 6 PM
+#   hubot ec2 unschedule <instance_id> [<instance_id> ...] Removes the Schedule tag from the instance(s)
+#
+# Notes:
+#   instance_id : [required] The ID of one or more instances to modify tags. For example, i-0acec691.
+#   --start : On a 24 hour clock, what hour of the day to tag the instance to be started.
+#   --stop : On a 24 hour clock, what hour of the day to tag the instance to be stopped.
+#   --start and --stop **only results in a tag being added to the instance**. Other automations are responsible for implementing stop and start
+
+cson = require 'cson'
+restrictor = require './restrictor'
+tags = require './tags'
+util = require 'util'
+ec2 = require('../../ec2.coffee')
+
+# 8 am to 6 pm
+DEFAULT_SCHEDULE_START = 8
+DEFAULT_SCHEDULE_STOP = 18
+
+createInstancesArray = (instance_ids) ->
+    instances = []
+    instance_ids = instance_ids.replace /^\s+|\s+$/g, ""
+    for i in instance_ids.split /\s+/
+      if i and not i.match(/^--/)
```

Comment:
This is picking up the time values, you could do something like `if i and not i.match(/^\-\-/) and i.length > 5`.